pyrtl/rtllib/multipliers.py

```diff
@@ -172,18 +171,20 @@ def signed_tree_multiplier(A, B, reducer=adders.wallace_reducer, adder_func=adde
             bits[i + j].append(x & y)
 
     res = reducer(bits, bits_length+1, adder_func)
```

Comment:
Oh wait I figured it out, shaved off the first bit, then multiplied, then extended the result and applied `_twos_comp_conditional()` and the result's bitwidth is `len(A) + len(B)`, which is good too